charms/worker/k8s/src/snap.py

```diff
@@ -272,7 +273,7 @@ def management(charm: ops.CharmBase) -> None:
     cache = snap_lib.SnapCache()
     for args in _parse_management_arguments(charm):
         which = cache[args.name]
-        if block_refresh(which, args):
+        if block_refresh(which, args) and not charm.is_upgrade_granted:
```

Comment:
Shouldn't this be `or`?

Reply:
This is not the case here. If the refresh is blocked (due to a different revision) but the upgrade machinery has verified and allowed the upgrade, using an `or` in this context will cause the block to execute every time, since one of the conditions (`block_refresh`) will always evaluate to true.

Reply:
@addyess Sounds good. I'll push the check into the block_refresh method.